scylla/tests/ccm-integration/cluster.rs

```diff
@@ -0,0 +1,832 @@
+use crate::logged_cmd::{LoggedCmd, RunOptions};
+use crate::node_config::NodeConfig;
+use anyhow::{Context, Error};
+use std::collections::{HashMap, HashSet};
+use std::fmt::Display;
+use std::net::{AddrParseError, IpAddr, Ipv4Addr, Ipv6Addr};
+use std::path::PathBuf;
+use std::process::Command;
+use std::str::FromStr;
+use std::sync::Arc;
+use tokio::fs::{metadata, File};
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::sync::RwLock;
+
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) enum DBType {
+    SCYLLA,
+    CASSANDRA,
+    DATASTAX,
+}
+
+#[derive(Debug, Clone)]
+pub(crate) struct ClusterOptions {
+    // Cluster Name
+    pub(crate) name: String,
+    // What to database to run: Scylla, Cassandra or Datastax
+    pub(crate) db_type: DBType,
+    // Scylla or Cassandra version string that goes to CCM.
+    // Examples: `release:6.2.2`, `unstable:master/2021-05-24T17:16:53Z`
+    pub(crate) version: String,
+    // CCM allocates node ip addresses based on this prefix:
+    // if ip_prefix = `127.0.1.`, then `node1` address is `127.0.1.1`, `node2` address is `127.0.1.2`
+    pub(crate) ip_prefix: NetPrefix,
+    // CCM does not allow to have one active cluster within one config directory
+    // To have more than two active CCM cluster at the same time we isolate each cluster into separate
+    // config director, each config directory is created in `root_dir`.
+    // Example: root_dir = `/tmp/ccm`, config directory for cluster `test_cluster_1` is going be `/tmp/ccm/test_cluster_1`
+    //  and cluster directory (that is created and managed  by CCM) for this cluster is going to be `/tmp/ccm/test_cluster_1/test_cluster_1`
+    pub(crate) root_dir: String,
```

Comment:
Again, ccm should be changed so that the commands can specify which cluster to operate on.

Reply:
true, but for now we have to dial with current behavior.